Rabbitmq volttron auth

shwethanidd opened the pull request:
# Description
Rearranged the code to encapulate all the RabbitMQ management code inside a class.

craig8 commented:
This pull request is now managed with CodeReviewHub. <a href='https://github.com/VOLTTRON/volttron/pull/1792'>Refresh</a>.<a href='#crh-update-none'></a>

schandrika commented:
1.  run-tests.sh 
lsb_release -a 
    is this needed or was it added initially just to find out the version of ubuntu running on travis

2. In aip.py
   line 427 - rmqmgmt class does not have a delete_rmq_user() method. 

3. control.py - 518
 line 518 - invalid 2nd parameter to function call

4. Most importantly - I am not able to start volttron with these changes. I get the error

 File "/home/vdev/git/myvolttron/volttron/utils/rmq_mgmt.py", line 938, in build_router_connection
    self.crts.create_ca_signed_cert(rmq_user, overwrite=False)
AttributeError: 'RabbitMQMgmt' object has no attribute 'crts'

shwethanidd commented:
1.   run-tests.sh
    lsb_release -a
    is this needed or was it added initially just to find out the version of ubuntu running on travis
Shwetha: Fixed
2.  In aip.py
    line 427 - rmqmgmt class does not have a delete_rmq_user() method.
Shwetha: Fixed
3. control.py - 518
    line 518 - invalid 2nd parameter to function call
Shwetha: Fixed
    Most importantly - I am not able to start volttron with these changes. I get the error

File "/home/vdev/git/myvolttron/volttron/utils/rmq_mgmt.py", line 938, in build_router_connection
self.crts.create_ca_signed_cert(rmq_user, overwrite=False)
AttributeError: 'RabbitMQMgmt' object has no attribute 'crts'
Shwetha: Fixed Typo. My bad. I tested the non-ssl version not the ssl implementation.

schandrika commented:
:shipit:
